compiler/rustc_session/src/options.rs

```diff
@@ -943,16 +943,30 @@ mod parse {
     pub(crate) fn parse_coverage_options(slot: &mut CoverageOptions, v: Option<&str>) -> bool {
         let Some(v) = v else { return true };
 
-        for option in v.split(',') {
-            let (option, enabled) = match option.strip_prefix("no-") {
-                Some(without_no) => (without_no, false),
-                None => (option, true),
-            };
-            let slot = match option {
-                "branch" => &mut slot.branch,
+        let set_branch_option = |slot: &mut CoverageOptions, option: &str| {
+            match option {
+                "no-branch" => slot.branch = false,
+                "branch" => slot.branch = true,
+                "mcdc" => {
+                    slot.mcdc = true;
+                    slot.branch = true
+                }
                 _ => return false,
-            };
-            *slot = enabled;
+            }
+            true
+        };
+
+        // Once an option is parsed we removed it from the array so that conflicting options such as "branch,no-branch" could be detected.
```

Comment:
Conflicting options should be allowed, with the last one winning.

That makes it possible to set a default value in one place, and then override the default value in another place. See https://github.com/rust-lang/compiler-team/issues/731 for more detail on why this is good.

For the same reason, we should allow `mcdc` and `no-mcdc` as options.